### ISSUE TITLE
Disabling lib gray out mods that uses that package

### ISSUE THREAD

**Issue by shkabo** [PrismLauncher/PrismLauncher#280]
### Role

modded MC

### Suggestion

In mods section, disabling lib mod should gray out mods that depends on it

### Benefit

Reduce launcher errors

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

I've started playing around with mods. A some mods have their own specific libs that they depend on. After removing certain mod, I have some libs in leftover that I'd like to remove and/or disable (why load them if they aren't being used?)
If I disable some lib, It should gray out text for mods that depends on them. E.g. disabling `Fabric Api` would disable most of the mods as most of them depends on it. Now I'm not sure you can distinguish libs from regular mods, but I believe it can be generated when getting mod metadata, and saving dependencies as a separate config file.

This is in some way related to #154 and in some minor way related to #116 and #37 (relationship with libs and dependencies)

I believe we should be able to get all of this data from the API

**Comment by ghost** [comment]
this should not affect the user's ability to launch the game as well as not give any annoying popups and only be done by scanning mod metadata. if so, it;s only for good.

**Comment by shkabo** [comment]
yes, this option should allow user to safely boot to the game without getting an error that dependency is missing if e.g. lib was disabled/removed. It should protect users in that way so they can always boot up into the game. On other hand it can help mod pack devs know which libs are needed for their mod pack w/o adding some libs that are maybe not needed, and people like me who like to experiment with different mods and who want to remove all leftovers from specific mod that's no longer needed by some other mod.
e.g. I've installed BetterEnd which requires bclib. I've uninstalled BetterEnd as I don't want to use it at the end, so I have leftover lib that's doing nothing. This lib should be grayed out disabled. Same thing if you e.g. removed bclib from your mods which is required by BetterEnd. In that case you'll end up with launcher error that lib is missing. With this functionality, BetterEnd will be grayed out/disabled and will let user start up game normally. Maybe some info window could be shown that specific lib was disabled due dependency with quick resolution to that issue (download missing lib) or leave it disabled like that

**Comment by ghost** [comment]
no please don't make this feature affect the way the game starts up, it should be only visual

**Comment by shkabo** [comment]
I'm jus thinkering options that could somewhat be available. It's up to the devs how to precisely implement such feature. So yea, discussion is highly welcome for this implementation :)

**Comment by JLUsr** [comment]
Based on the comments it sounds like there is some debate regarding having this feature be just a visual or something proactive, IMO it should work something like this.
1. I've disabled a library mod (let's imagine it's called **libMod**) that **Mod-A** and **Mod-B** require to run
2. A pop-up shows up with text along the lines of _"**libMod** is a dependency needed for **Mod-A** and 1 other to work properly"_
3. At this point there would be choices like `Always disable dependents when disabling libMod`, `Disable dependents` (this one would be a one-time action), and `Ignore` _Edit1: the indication here is that it would persistently ignore until it has been recognized that the mod has been enabled, and then disabled again, at which point it would show the prompt again._

Regardless of which choice you make all dependants would have some way to indicate that their required dependency is disabled (my vote for that would be a `❗` next to the mods name that would state something like _"The dependency **libMod** has been disabled"_ when you hover over it)

Inversely, when you disable a dependent, have the lib mod that it normally depends on show a `⚠️` next to it. In the case of disabling **ModA** nothing would happen; however, if you disabled both **ModA** and **ModB** the icon would show up next to **libMod** and state something like _"There are no required dependents for **libMod** enabled"_ when you hover over it. (no pop-ups for this since there are plenty of times a mod doesn't specifically require another and can be used optionally)

Edit2: A similar solution that would work without a pop-up is to keep the `❗` and `⚠️` notification system and make it so that a `❗` shows up next to the lib mod that was disabled and allow for it to be clicked to disable all of the dependants that require it.

**Comment by ghost** [comment]
This is not required. Just show a "!" next to mods that would not work, and a warn in the console. I don't want the launcher to be bloated with any popups of information generally known by the user. And this check should be toggleable.

> Edit2: A similar solution that would work without a pop-up is to keep the ❗ and ⚠️ notification system and make it so that a ❗ shows up next to the lib mod that was disabled and allow for it to be clicked to disable all of the dependants that require it.

If you mean what I think you do this would be a good solution.
